### PR TITLE
feat: add sklearn-style input validation to LLMFeatureEngineer

### LINKED ISSUE #34: sklearn-style input validation to LLMFeatureEngineer
## Summary

`LLMFeatureEngineer` currently performs no input validation at any of its public method boundaries. Users who pass wrong types, mismatched shapes, empty DataFrames, or invalid constructor parameters receive either a confusing internal error deep in the call stack, silent wrong behaviour, or no error at all.

Following the conventions established by scikit-learn, every public-facing method should validate its inputs as early as possible and raise a `ValueError` with a descriptive message. This makes the library easier to debug, safer to use, and more consistent with the rest of the sklearn ecosystem it integrates with.

Specifically, `fit()` and `fit_selective()` should validate that `X` is a non-empty `pd.DataFrame` and that `y`, when provided, is a `pd.Series` of the same length. They should also store `n_features_in_` and `feature_names_in_` as fitted-state attributes — the standard sklearn convention — so that `transform()` can detect at call time when a DataFrame is missing columns that were present during training. This is a particularly important footgun today: passing a test set with a renamed or dropped column silently produces incomplete output rather than an error.

Constructor parameters (`max_features`, `verbose`) should also be validated eagerly so that misconfiguration is caught at object creation rather than at `fit()` time. Similarly, `fit_selective()` should reject nonsensical values like `n_rounds < 1` and malformed `eval_set` tuples. Finally, `evaluate_features(is_transformed=True)` should verify that the expected generated feature columns are actually present in the passed DataFrame — currently a missing column causes a silent `KeyError` or `NaN` metrics with no explanation.

## Acceptance criteria

- [ ] All new validators raise `ValueError` with descriptive messages
- [ ] `n_features_in_` and `feature_names_in_` are set after `fit()` and `fit_selective()`
- [ ] `transform()` raises `ValueError` when passed a DataFrame missing fit-time columns
- [ ] Tests added for each new validation path
- [ ] Existing tests pass unchanged

### PR DESCRIPTION
**Pull request by P-r-e-m-i-u-m**
Closes #34

## Summary
Added sklearn-style input validation to all public method boundaries in `LLMFeatureEngineer`.

### `skfeaturellm/feature_engineer.py`
- `__init__()`: Validates `max_features` (positive int or None) and `verbose` (non-negative int)
- `fit()`: Validates `X` is non-empty DataFrame, `y` is Series of same length. Stores `n_features_in_` and `feature_names_in_`
- `transform()`: Validates `X` is DataFrame and raises `ValueError` if columns present during fit are missing
- `fit_selective()`: Validates `X`, `y`, `n_rounds >= 1`, and `eval_set` tuple format. Stores `n_features_in_` and `feature_names_in_`
- `evaluate_features(is_transformed=True)`: Raises `ValueError` if expected generated feature columns are missing

### `tests/test_feature_engineer.py`
- Added 11 new tests covering all new validation paths
- All existing tests pass unchanged

### CODE REVIEW

File: skfeaturellm/feature_engineer.py [hunk 1]
```diff
@@ -87,18 +97,32 @@ def fit(
         self : LLMFeatureEngineer
             The fitted transformer
         """
+        if not isinstance(X, pd.DataFrame):
```
Comment (RobertoCorti) [inline skfeaturellm/feature_engineer.py]: Can you abstract this valudation logic into a standalone `validate_data()` function in [`utils.validation`](https://github.com/RobertoCorti/skfeaturellm/blob/main/skfeaturellm/utils/validation.py)? Similar to how scikit-learn handles it (see [here](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/ensemble/_forest.py#L335)), this would keep `fit()` cleaner and make the validation reusable across other methods/classes down the line.

File: skfeaturellm/feature_engineer.py [hunk 2]
```diff
@@ -125,20 +148,21 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         pd.DataFrame
             Input dataframe with the generated features
         """
-        # if fit has not been called, raise an error
         check_is_fitted(self)
-
-        # Convert LLM output to executor config and apply prefix to feature names
+        if not isinstance(X, pd.DataFrame):
```
Comment (RobertoCorti) [inline skfeaturellm/feature_engineer.py]: same comment as [here](https://github.com/RobertoCorti/skfeaturellm/pull/35/changes#r2909842080). A general `validate_data` can be used also here and avoid possible duplications

File: skfeaturellm/feature_engineer.py [hunk 3]
```diff
@@ -229,12 +253,43 @@ def fit_selective(  # pylint: disable=too-many-arguments
             The fitted transformer. Call ``transform()`` to apply the selected
             features and ``to_transformer()`` to export them for production.
         """
+        if not isinstance(X, pd.DataFrame):
```
Comment (RobertoCorti) [inline skfeaturellm/feature_engineer.py]: same comment as [here](https://github.com/RobertoCorti/skfeaturellm/pull/35/changes#r2909842080). A general `validate_data` can be used also here and avoid possible duplications

**Review by RobertoCorti — CHANGES_REQUESTED** [review]

**Comment by RobertoCorti** [comment]
It looks like pre-commit is failing in CI. Could you install it locally and run it before pushing?

Use the command

```bash
poetry run pre-commit install
```

After that, it'll run automatically on each commit and catch any issues before they hit CI.

**Comment by P-r-e-m-i-u-m** [comment]
"Refactored validation logic into a standalone validate_data() function in utils/validation.py and updated fit(), transform(), and fit_selective() to use it. Ready for re-review @RobertoCorti

File: skfeaturellm/utils/validation.py [hunk 4]
```diff
@@ -24,3 +24,49 @@ def _is_fitted(estimator):
...
+        Name of the estimator, used in error messages.
+
+    Raises
+    ------
+    ValueError
+        If X is not a non-empty DataFrame, or X and y have different lengths.
+    TypeError
+        If X is not a DataFrame or y is not a Series.
+    """
+    import pandas as pd
```
Comment (RobertoCorti) [inline skfeaturellm/utils/validation.py]: Please move `import pandas` as pd to the top of the file, outside the function.

**Comment by RobertoCorti** [comment]
Again, it looks like that CI/CD is failing. Install pre-commit checks through:

```python 
poetry run pre-commit install
```

them to run the pre-commit checks execute

```python
poetry run pre-commit run --all-files
```

**Review by RobertoCorti — CHANGES_REQUESTED** [review]
changes looks good, please fix my only comment and ci/cd

**Comment by P-r-e-m-i-u-m** [comment]
"Moved import pandas as pd to the top of validation.py. Pre-commit should pass now. Ready for re-review @RobertoCorti

**Comment by P-r-e-m-i-u-m** [comment]
"Fixed feature_names_in_ missing in transform() — added fallback to set it from X.columns if not present. Ready for re-review @RobertoCorti 👍"
